lib/samson/hooks.rb

```diff
@@ -30,6 +32,24 @@ def add_migrations
         migrations = File.join(@folder, "db/migrate")
         Rails.application.config.paths["db/migrate"] << migrations if Dir.exist?(migrations)
       end
+
+      def add_lib_path
+        engine.config.autoload_paths += Dir["#{engine.config.root}/lib/**/"]
+      end
+
+      def add_decorator(klass_name)
+        return if klass_name.nil?
+        decorator_file = engine.config.root.join("app/decorators/#{klass_name.downcase}_decorator.rb")
```

Comment:
So use the same mechanism (the ```model_defined``` callback) apart from the fact that is going to be defined in models we are going to re-open a class?